cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/NodeBuilder.kt

```diff
@@ -269,3 +270,53 @@ fun <T : Node, S> T.codeAndLocationFrom(frontend: LanguageFrontend<S, *>, rawNod
 
     return this
 }
+
+fun <T : Node, S> T.codeAndLocationFromChildren(
+    frontend: LanguageFrontend<S, *>,
+    parentNode: S
+): T {
+    var nodesWithLocation =
```

Comment:
Why don't you use `var nodesWithLocation = this.allChildren<Node>{it.location != null && it.location?.region != Region()}` instead of making this worklist and while loop?

Reply:
I found a comparison function of single elements with comparator function that allowed me to implement the lexicographic order. So I managed to make it more efficient:
 - no sorting
 - comparing every node only once